### DISCUSSION TITLE
RFC: Medication Terminology

### DISCUSSION THREAD

**Discussion by JensPenny** [hl7-be/medication#225] [Ideas]
The purpose of this document is to provide guidance for the medication levels that we want to support in a medication terminology. 

This document will first give an overview of the samv2 concepts that seem most useful to encode, along with a proposal for an URI to identify these
Then, we will go over a couple of existing (proposals for ) profiles, to indicate how we can and should restrict the possible medication concepts that are used for a certain profile.
Lastly there is a proposal to add more Samv2 concepts that are not encompassed in this document. We propose a basic function to create a medication-terminology-URI that can be used for more keys. 

## Contributing
Just add your notes to the discussion below.
**Open questions** contains questions that will be touched in the discussion. You can add more questions in the discussions.
**Example use cases** could use more examples and profiles that require medication. The goal is to find a specific profile where the medication can not be represented by the proposed levels.

## General guidelines
- An identifier must be used for one **concept**. 
	- The concept in terminology can have extra information
- The usage of an identifier of a concept will always refer to the **currently active** version in the terminology
- If a version of a concept based on a date needs to be retrieved at a later time, this filtering should happen by querying the terminology server for the **concept identifier** together with a **validOn** date.

The medication identifiers will be encoded in a [Coding](https://www.hl7.org/fhir/datatypes.html#Coding).
- The system **must** always be prefixed with a common identifier. The second part of the system will annotate the level. 
- The version **can** be added to the coding object. If the version is missing, the last version will be assumed. 
- The system has the following structure:
	- `urn:be:samv2:`
	- The structure is scheme: urn, path: be:samv2. 
	- For the path we try to add the **authority** that maintains samv2, and then the actual samv2-name to indicate that this is the samv2 database

### AMPP - Actual Medicinal Product Package
**Unique identifier**: CTI-extended
**Proposed system**: `urn:be:samv2:amppackage` . This is not ampp to enhance the difference with the amp-identifier.
**Example value**: [200697-05](https://samviewer.digile.be/nl/sam/ampps/200697-05)

This represents an actual packaging level. 
This is useful in a lot of contexts, since this is the lowest, tangible level that a single product packaging can have. 
This is also a concept that has value in logistical product chains, because it is actually deliverable.

There **could** be an option to add another namespace for cnk to this level, but I would personally not do that and eventually provide a way to go from DMPP to AMPP, or resolve this when searching for the terminology.
### AMP - Actual Medical Product
**Unique identifier**: code
**Proposed system**: `urn:be:samv2:amp` **caution-low difference with ampp**
**Example value**: [SAM662104-00](https://samviewer.digile.be/nl/sam/amps/SAM662104-00)

This represents a unidose for an actual product. 
While most systems work with direct packaging, this is useful for several logistical systems where people work with separate pills. 
For example: hospitals
### VMP group - virtual medicinal product group
**Unique identifier**: code
**Proposed system**: `urn:be:samv2:vmpgroup` 
**Example value**: [24745](https://samviewer.digile.be/nl/sam/vmp_groups/24745)

This represents the (prescribeable) substance level.
This is a special level in Belgium that describes a group of VMP's that can be used to, for example, fill in a prescription. 
### VMP - virtual medicinal product
**Unique identifier**: code
**Proposed system**: `urn:be:samv2:vmp`
**Example value**: [26484](https://samviewer.digile.be/nl/sam/vmps/26484)

The virtual product. Contains substance + doseform (ex.`paracetamol 1 g granulaat (or.)`).
This is an intermediary level that bridges VMP group and AMP. 
### Nonmedicinal products
**Unique identifier**: cnk
**Proposed system**: `urn:be:samv2:nonmedicinal`
**Example value**: [1182211](https://samviewer.digile.be/nl/sam/non_medicinal_products/1182211)

These products are nonmedicinal product packages that can also be sold and dispensed by a pharmacy. These products share the same cnk-namespace that **dmpp's** have as well.

### Magistral preparations
For these preparations another specialized discussion is needed.
There are compounding ingredient tables and compounding formula tables that might be usable for a magistral preparation, but I have not researched this deeply. 

For me (Jens) this does not feel like terminology, but a profile for the medication that is custom prepared. For the ingredient level the **VMP** looks sufficient.
The only issue is that product strength is a field on the AMP, not on the VMP. 
There is another complexity in the fact that a magistral preparation sheet (what is prescribed) can have a different setup as what is actually prepared (by the pharmacist). The prescription can use virtual products, while the preparation can use actual substances or products.

There could be two major namespaces for magistral preparations:
- A namespace that references a standardized preparation, maintained by APB (is this correct?)
- A generic medication concept that gets it's own uuid and that is represented in terminology by a list of substances, their respective doses and the doseform of the magistral preparation.

## Moving between levels
The basic structure of SAM means that there are two levels. 
- The actual level, that contains actual names of products
- The virtual level, that focusses on substance. 

In most cases, one level represents a list of lower levels. In general you can traverse most of the useful levels with this structure. 
- **vmp group** represent a list of **vmp's**
- **vmp** represents a group of **amp's**
- **amp** represents a group of **ampp's**
- An **ampp** contains a group of **dmpp's**. The dmpp contains a cnk code used to identify medication in earlier systems (KMEHR).

## Other identifiers
- Other identifiers **should** also use the system prefix `urn:be:samv2:`
- Other identifiers **should** use the table name that best describes the full context as postfix in the system.

### VTM - Virtual Therapeutic Moeity
The proposal does not include this level, since this is not really required to traverse a tree.
If this is required in a level (for instance magistral preparations), it can be included. 
A VTM represents a list of VMP's for that substance. This is comparable to a vmp-group, but since the group is actually used in practical use cases, that is preferred.

### VMP component and other components
We propose to include these levels in the vmp and amp terminologies.
### CNK for medication (DMPP)

The proposal is to **not** use this as a separate level, but as a property of the **AMPP** level. 

### ATC code
While ATC is a useful **classification**, it **should not** represent an actual product. 
ATC is a property of other concepts. In this case: AMPP.

### Snomed codes
Snomed has useful concepts for several of the levels that are contained in samv2. Where possible, there can be a bridge between (one or more) samv2 concepts and snomed, but that is at this moment out of the scope of this document. 
[This nhs link](https://www.nhsbsa.nhs.uk/pharmacies-gp-practices-and-appliance-contractors/dictionary-medicines-and-devices-dmd) explains how the UK did this. The UK is a good reference here, since they use a simplified version of the samv2 data model. 

## Example use cases
This section describes some example use cases, and the concepts that might be useful in these contexts. 
These are ordened by importance. 
### Medication dispense
- **AMPP** for actual packaging
- **Nonmedicinal** for supplements 
- **magistral preparations** for custom preparations and generics prepared by the pharmacy
### Medication therapy line
- **AMPP** for inquiries on what the patient is taking, and filling in the patient file based on this level
- **AMP** if the patient only remembers the basic type of pills
- **VMP** for generics. **VMPgroup** is also an option here
- **nonmedicinals** for supplements
### Vaccination
- **AMPP** for the actual medication administration
	- **AMP** might suffice at this level
- **snomed** indication might suffice here as well, if the actual product is not known
### Allergy / Intolerance
- **AMP** for an intolerance to a product
- **VTM** (not included) for an allergy based on a substance
	- Might be **substance** as well
- **snomed** for allergies to for ex. sunlight, latex

## Open issues
- Sam v2 has versions 4, 5 and 6. It is unclear to me how we should segment these versions. 
	- The current proposal puts the version tin the `Coding`-segment
	- This should allow us to differentiate between the versions that are contained in the terminology server
- For allergies: how do we handle a 'kruisallergie' (cross-reactivity).
- ...
## Other countries
- UK - NHS
	- The system is [https://dmd.nhs.uk](https://dmd.nhs.uk)
- See  [HL7 Europe inventory of medication concepts](https://docs.google.com/spreadsheets/d/1NZeUnFV4HUn471ccao5gooym72RbHqOXaYjiNujKDi4/edit?gid=0#gid=0) for an overview
## Fhir relevant information
**coding.system** rules
```
The URI may be an OID (urn:oid:...) or a UUID (urn:uuid:...). OIDs and UUIDs SHALL be references to the HL7 OID registry. Otherwise, the URI should come from HL7's list of FHIR defined special URIs or it should be an absolute reference to some definition that establishes the system clearly and unambiguously.
```

**coding.version** rules. 
```
Where the terminology does not clearly define what string should be used to identify code system versions, the recommendation is to use the date (expressed in FHIR date format) on which that version was officially published as the version date.
```
Do note that I would like to change this to the version of sam used (eg. 5 or 6).
An alternative is `version:samrelease`, which would also point to a single identifiable version of sam. For example: `4:9700`, `5:9701` or `6:9699`.
## Other links
- [Github issue HL7-be](https://github.com/hl7-be/medication/issues/222): This issue is the original start of this document
	- The github references it, but the [UK dm+d guide](https://digital.nhs.uk/services/terminology-and-classifications/dm-d#implementing-dm-d) is a very good guide for a lot of the concepts that we try to introduce.
- [HL7 Europe inventory of medication concepts](https://docs.google.com/spreadsheets/d/1NZeUnFV4HUn471ccao5gooym72RbHqOXaYjiNujKDi4/edit?gid=0#gid=0): This contains an inventory of other international concepts. This can be used as a guideline to see what the important concepts are.
- [Sam viewer](https://samviewer.digile.be/nl/sam): Quick lookup for sam concepts
- [Personal sam to sqlite project](https://github.com/JensPenny/SamToSqlite/blob/master/src/main/kotlin/db/ActualMedicineSamTableModel.kt) for a quick overview of the used tables and their unique identifiers
- [Samportal](https://www.samportal.be/nl/sam) for the base documents used in sam
- Earlier discussions on the Belgian level are included as well
	- [Fhir scenarios for dispenses](https://docs.google.com/document/d/1rTGHvdNMZfR4MZvVVIfUnoAi8eDQsYTpcKBB3B8q4JI/edit?usp=drive_web&ouid=100073800108979558136) to double-check the examples of medication terminology that should be used for dispenses
	- [Concepts that should be used in a medication scheme](https://docs.google.com/document/d/1rTGHvdNMZfR4MZvVVIfUnoAi8eDQsYTpcKBB3B8q4JI/edit?tab=t.0#heading=h.h7vvfbjcot2s) for the earlier discussion on what concepts that are used
- [URI RFC 3986](https://datatracker.ietf.org/doc/html/rfc3986#section-1.1.1)

**Comment by costateixeira** [comment]
why urn:smals:samv2:amp and not urn:be:samv2:amp ? why smals?

**Reply by JensPenny** [reply]
I initially used smals here as the 'authority' for the URI, but in this context urn:be:samv2 or even urn:be:medication:samv2 might make more sense. At least in international context this will be more clear.

**Reply by JensPenny** [reply]
I have changed this information in the posted proposal

**Comment by ildossche** [comment]
Thanks Jens for pulling this together! In the section on nonmedicinal products "dmpp's" are mentioned. Should this be ampp?

**Reply by JensPenny** [reply]
This may just be a small semantic issue.
The cnk-code for medication is not defined on the AMPP level, but on the DMPP level. This is the delivery environment for the actual package. It is possible that a 2 cnk's point to the same ampp, but that the cnk differs because the package for hospital environments is different then the cnk for packages for the public or ambulatory environment.

An example that will hopefully clarify this is [crestor](https://samviewer.digile.be/nl/sam/ampps/250223-05). 
![image](https://github.com/user-attachments/assets/e9af91a4-4b83-4087-b846-5088bbb10997)

I'll see if I can clarify this, either in the **nonmedicinal**-section or in the **moving between levels**-section.

**Reply by JensPenny** [reply]
I added the following:

> - An **ampp** contains a group of **dmpp's**. The dmpp contains a cnk code used to identify medication in earlier systems (KMEHR).

**Comment by JensPenny** [comment]
## Synopsis of the meeting at 2024/12/17

- Jens Penny: presenter
- Ilse Dossche: sciensano (Barometers and registers)
- Bart Rondou: smals (recip-e)
- Jose Costa Texeira: HL7 Belgium 

### Changes

- the identifier for ampp is changed to `urn:be:samv2:amppackage `.

### Discussions

- `ampp` is also usable as a logistical concept, since it represents an actual packaging. This is the big differentiator of this level with other concepts. 
- magistral preparations come in three forms:
  - an ad hoc preparation based on free text, with the preparation steps as plaintext.
  - an ad hoc preparation where the components are coded concepts (`vmp` / `vtm` + dose?)
    - These concepts are listed in SAM and should be used from here.
  - officinal preparations: basic preparations that are defined.
- magistral preparations do need more specialized discussion with other groups (pharmacists) to sketch out the full concept
  - There is a difference between the steps of the preparation, and the dose that needs to be taken/applied as well

- Other usable concepts are `snomed`
  - Ilse points out that most current FHIR structures use snomed for medication concepts (like vaccineCode for BeVaccination)
  - The barometer-registers will go towards snomed as well
  - Jens: snomed is fine, but a lot of other information is already encoded in sam. A mapping may be required later.
  - Jose: try to avoid mappings by splitting of attributes of medication in samv2 that are easily encoded in different properties that map to snomed organically (like route of administration / dose forms / ...)
  - Jens: some kind of way to navigate from sam-concept to snomed will be required either way, or some guidance on how to map this. Further discussion on how this should be used is needed.

- Allergy-intolerance: are amp's required (isn't snomed enough)
  - EHR's will allow you to record an intolerance for a specific product
  - The question becomes: at what level are you allergic/intolerant
  - Another case for better mappings / documentation for when and how we use snomed in existing systems

- Snomed as system
  - Snomed is a good choice for standardization
  - Snomed can not replace medication (amp / ampp) in its current form.
  - guidelines needed on how we bridge this gap between registration (based on medication) and reporting (based on snomed).

### Questions

- Is it possible to add a view on what the extra information is for the different levels
  - For example: what does an `ampp` add to an `amp`? Is it just dose-quantity and packaging information?
  - Jens: will check, but this isn't that easy since samv2 isn't the easiest format. The fact that it is in essence an xml with separate levels can make this a bit easier (or at least possible).
- Can you add an example for medicationPrescription (recip-e) next to medicationDispense
  - Jens: will do. More examples may be proposed / added at will as well

**Comment by KarlienHL7Belgium** [comment]
a diagram that shows the links b/w the concepts should ideally be added here

**Reply by JensPenny** [reply]
## WG medication 2025/01/07

- @JensPenny will create a samv2 simplified diagram to visualize the way on how the different levels will interact. This will go in 'moving between the levels'
- @JensPenny will tabularize the examples, so it's more clear on what the proposed types of medication / fhir profile contain.
- @JensPenny will edit the implementation guide for a profile (medicationline?) to refer to the different types of medication that a medication-resource can represent for this profile. See #231
  - @costateixeira : if you can provide a MR / diff to a code point where this is done earlier, or just can walk me through on how to update the IG 

There is an open issue on who will actually provide the terminology. This is not something that I can personally do.

**Comment by costateixeira** [comment]
Can we add an example or two for a medication along the different levels of granularity?
This would help with the navigation / guidance and creating examples.

**Reply by JensPenny** [reply]
Planned by @JensPenny at 10/02/2025.

**Comment by smals-hej** [comment]
A question on the unique identifier that is proposed for AMPP: why was CTI-Extended chosen instead of CNK code?

**Reply by JensPenny** [reply]
There are a couple of reasons, but mainly because the documentation for the [samv2 logical model](https://www.samportal.be/nl/sam/documentation) points to the CTI-extended as the semantic key. This is the key that is also used in the other tables that reference the AMPP.

**Comment by michel-demey** [comment]
Hi all. What is the next step here, to switch from proposal to effective ? We are waiting to start building interesting things on this...

**Comment by KarlienHL7Belgium** [comment]
@costateixeira , @JensPenny : can we take this up in our next WG meeting on 2 June ? Michel has asked for an update

**Comment by costateixeira** [comment]
yes

**Comment by maximesteisel** [comment]
@costateixeira as discussed at the FHIR-a-thon, here is a summary of the different concepts, what they include and what Jens has proposed as the code system

| Sam concept | Meaning | Exemple | Code system |
|--------|--------|--------|--------|
| VTM |  active ingredient | [acide acétylsalicylique ¦ SAM viewer](https://samviewer.digile.be/fr/sam/vtms/15) | N/A (not often used in practice) |
| VMP-group | VTM + route of administration (oral, IV, …) +dosage | [acide acétylsalicylique oral 100 mg ¦ SAM viewer](https://samviewer.digile.be/fr/sam/vmp_groups/307) |  urn:be:samv2:vmpgroup |
| VMP | VMP-group + galenic form (pill, powder, …) | [acide acétylsalicylique 100 mg  powder ... ¦ SAM viewer](https://samviewer.digile.be/fr/sam/vmps/398) | urn:be:samv2:vmp |
| AMP | VMP + commercial name | [Aspegic 100 mg sol. buv. (pd... ¦ SAM viewer](https://samviewer.digile.be/fr/sam/amps/SAM120617-00) | urn:be:samv2:amp |
| AMPP | AMP + number of doses per box | [Aspegic 100 mg sol. buv. (pd... ¦ SAM viewer](https://samviewer.digile.be/fr/sam/ampps/120617-02) | urn:be:samv2:amppackage | 
| DMPP | AMPP + delivery environment (hospitalized patient vs ambulatory patient in hospital vs public pharmacy | [public: CNK 1652-049 ¦ SAM viewer](https://samviewer.digile.be/fr/sam/dmpps/33434) |  NA (model it as AMPP + a separate delivery environment property) | 
| other product | non-drug products (food, cosmetics, ...) |  [Isostar high energy banaan 40g ¦ SAM viewer](https://samviewer.digile.be/nl/sam/non_medicinal_products/1182211) | urn:be:samv2:nonmedicinal | 

For our project at UCL we will use those three URI:

urn:be:samv2:vmpgroup
urn:be:samv2:vmp
urn:be:samv2:amp

It would be most helpful if those URIs could be published officially so that we don't have base our system on de-facto standards.

If you need more explaination about how the SAM works I would be happy to have a meeting with you or you can also contact the team at RIZIV-INAMI who maintains it (CTGCRM database (RIZIV-INAMI) <[CTGCRM.database@riziv-inami.fgov.be](mailto:CTGCRM.database@riziv-inami.fgov.be)>)
